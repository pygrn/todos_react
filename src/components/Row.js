import React from 'react'
import PropTypes from 'prop-types'

const Row = ({todo}) => (
  <tr>
    <td>{todo.description}</td>
    <td>{todo.due_date}</td>
  </tr>
)

Row.propTypes = {
  todo: PropTypes.shape({
    pk: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    due_date: PropTypes.string.isRequired
  })
}

export default Row