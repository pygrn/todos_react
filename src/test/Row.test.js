import React from 'react'
import Row from '../components/Row'
import renderer from 'react-test-renderer'


const row = {
    pk: 1,
    description: 'Todo 1',
    due_date: '2017-11-08T21:01:35Z'
}

test('Row', () => {
    const component = renderer.create(<Row todo={row}/>)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})