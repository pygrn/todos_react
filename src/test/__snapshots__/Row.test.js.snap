// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row 1`] = `
<tr>
  <td>
    Todo 1
  </td>
  <td>
    2017-11-08T21:01:35Z
  </td>
</tr>
`;
